ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala

```diff
@@ -475,12 +526,13 @@ class TransactionService(session: LedgerSession) extends LedgerTestSuite(session
       "TXNotDiscloseCreateToNonChosenBranchingController",
       "Not disclose create to non-chosen branching controller") { context =>
       for {
-        ledger <- context.participant()
-        Vector(alice, bob, eve) <- ledger.allocateParties(3)
+        Vector(alpha, beta) <- context.participants(2)
+        alice <- alpha.allocateParty()
+        Vector(bob, eve) <- beta.allocateParties(2)
         template = BranchingControllers(alice, false, bob, eve)
-        create <- ledger.submitAndWaitRequest(alice, template.create.command)
-        transaction <- ledger.submitAndWaitForTransaction(create)
-        transactions <- ledger.flatTransactions(bob)
+        create <- alpha.submitAndWaitRequest(alice, template.create.command)
+        transaction <- alpha.submitAndWaitForTransaction(create)
+        transactions <- beta.flatTransactions(bob)
```

Comment:
This test could succeed even with a faulty disclosure mechanism, simply by way of `beta.flatTransactions` always being called before `beta` would receive the transaction.
To make sure that we can make a negative statement (transaction X created on alpha is not seen by party Alice), we need to ensure that party Alice does see a later transactions. Only then can we be sure that party Alice has seen all actions that happened on alpha.

Reply:
Addressed by https://github.com/digital-asset/daml/pull/2900/commits/886f7e81b2be070d2ca25f98e29079995fd0a3b2